pkg/network/dhcp/configurator.go

```diff
@@ -9,15 +9,15 @@ import (
 	netdriver "kubevirt.io/kubevirt/pkg/network/driver"
 )
 
-const dhcpStartedDirectory = "/var/run/kubevirt-private"
+const defaultDHCPStartedDirectory = "/var/run/kubevirt-private"
 
 type Configurator struct {
 	advertisingIfaceName string
 	cacheFactory         cache.InterfaceCacheFactory
 	filterByMac          bool
 	handler              netdriver.NetworkHandler
 	launcherPID          string
-	dhcpStartedDirectory string
+	DHCPStartedDirectory string
```

Comment:
I don't think the podNIC_test should invoke the actual code of the dhcp configurator. It should use a mock.
If we will use a mock, I believe this change will be redundant.

Reply:
I'm replying to this because I constantly see your requests for relying on mocking. 

I'm aware these are unit tests; never the less, the only behavior in the `DhcpConfigurator` that requires "mocking" is where to put the file representing the lock. And that's because we don't have a clear way of exporting a constructor to be used just for testing.

It's more work, and adds overhead to the tests (since you'll have to add the N mock calls) mocking rather than simply calling the structure. Furthermore, you'll have to define every data structure as an interface in order to mock it - just for sake of enabling unit testing to use mocks. E.g. your production code will change because of how your tests are designed. (I'm aware we're exporting a member just for that... ).

I just want to know where we'll be going for the future. Do it the java way and mock everything ?.. I hope not.

Reply:
Using mocks here allow us to keep this field private (that's good) so I think it's worth it.